fixtures/test-fixtures.sh

```diff
@@ -24,7 +27,9 @@ run_fixture() {
 
 main() {
   find * -maxdepth 0 -type d -print0 | while IFS= read -r -d '' fixture; do
-    run_fixture "$fixture"
+    # Since we are creating a subshell, all env vars created by rp_env will be lost
+    # Return code is preserved
+    (run_fixture "$fixture")
```

Comment:
Is there any side effects of this? Are we sure that the return code is preserved?

I'd test it manually with a failing fixture on purpose to see if the behavior is the same.

also nit:

```suggestion
    # Since we are creating a subshell, all environment variables created by rp_env will be lost
    # Return code is preserved
    (run_fixture "$fixture")
```

Reply:
Interesting. I would have used `custom_env_teardown()` and added it to `run_fixture()` before retuning.

But if the subshell works, it's certainly a better option.

Reply:
I echo'd the return value during my modifications and it did preserve the return code properly. As for any side effects I couldn't notice any.

<details><summary>Fixtures run normally:</summary>
<p>

![image](https://github.com/user-attachments/assets/b3796f7a-d794-45c2-92c0-b9668d0c6cfb)
</p>
</details> 

<details><summary>Modified `test` to fail</summary>
<p>

![image](https://github.com/user-attachments/assets/57346884-5545-4e35-8b73-d7bda4553439)

</p>
</details>